src/go/agwd/config/config.go

```diff
@@ -193,3 +197,9 @@ func LoadConfigFile(cm *ConfigManager, path string) error {
 	cm.Merge(loaded)
 	return nil
 }
+
+// UpdateConfig updates CongfigManager with a config file.
+func (cm *ConfigManager) UpdateConfig(config *config.AgwD) error {
```

Comment:
should we just add `Merge` to the `Configer` interface instead? Or did you intend to add validity/error checks here? (if the latter, let's add a TODO comment)

Reply:
also: add unit test pls

Reply:
exactly - figured in the future we would we would break up the api a bit or limit what fields users can update etc. i'll add a todo.